src/components/shapes/attributes.js

```diff
@@ -224,6 +225,50 @@ module.exports = templatedArray('shape', {
             '`config.editable` or `config.edits.shapePosition`.'
         ].join(' ')
     },
-
+    label: {
+        text: {
+            valType: 'string',
+            dflt: null,
+            editType: 'arraydraw',
+            description: 'Sets the text to display with shape.'
+        },
+        font: fontAttrs({
+            editType: 'calc+arraydraw',
+            colorEditType: 'arraydraw',
+            description: 'Sets the shape label text font.'
+        }),
+        position: {
+            valType: 'enumerated',
+            values: [
```

Comment:
It looks like we need to have `auto` `dflt` and `value` in respect to `src/components/shapes/defaults.js` logic.

Reply:
@archmoj What would be the use case for an `auto` value for `textposition`?

Reply:
> @archmoj What would be the use case for an `auto` value for `textposition`?

Hmm.. that's a good idea. But as far as I recall we don't have such an option for `textposition` in other places. So perhaps we could keep this part as is.